test/e2e/suite-plugins/consumer-restriction.go

```diff
@@ -0,0 +1,260 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package plugins
+
+import (
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/apache/apisix-ingress-controller/test/e2e/scaffold"
+)
+
+func CreateBasicAuthConsumer(name, username, password string) string {
+	ac := fmt.Sprintf(`
+apiVersion: apisix.apache.org/v2beta3
+kind: ApisixConsumer
+metadata:
+  name: %s
+spec:
+  authParameter:
+    basicAuth:
+      value:
+        username: %s
+        password: %s
+`, name, username, password)
+	return ac
+}
+
+func Authentication(basicAuthInfo interface{}) string {
+	basicAuth, ok := basicAuthInfo.(map[string]interface{})
+	if !ok {
+		return ""
+	}
+	username, _ := basicAuth["username"].(string)
+	password, _ := basicAuth["password"].(string)
+	str := username + ":" + password
+	return "Basic " + base64.StdEncoding.EncodeToString([]byte(str))
+}
+
+//consumer-restriction plugin testing
+var _ = ginkgo.Describe("suite-plugins: consumer-restriction plugin", func() {
+	opts := &scaffold.Options{
+		Name:                  "default",
+		Kubeconfig:            scaffold.GetKubeconfig(),
+		APISIXConfigPath:      "testdata/apisix-gw-config.yaml",
+		IngressAPISIXReplicas: 1,
+		HTTPBinServicePort:    80,
+		APISIXRouteVersion:    "apisix.apache.org/v2beta3",
```

Comment:
Should we use `apisix.apache.org/v2` now? cc @tao12345666333

Reply:
> Should we use `apisix.apache.org/v2` now? cc @tao12345666333

good catch! @AlinsRan please change this to `apisix.apache.org/v2` 
thanks!

Reply:
why? How are they diff? Does this need to be done for subsequent test cases.

Reply:
@tao12345666333

Reply:
> why? How are they diff? Does this need to be done for subsequent test cases. 

v2 is the latest stable version. v2 is also required in subsequent test cases

Reply:
> > why? How are they diff? Does this need to be done for subsequent test cases.
> 
> v2 is the latest stable version. v2 is also required in subsequent test cases

ok

Reply:
In fact, we haven't released the v2 version of CRD, so there is no need to modify it here. 😅

@gxthrj @AlinsRan